zerver/migrations/0395_add_function_process_rendered_content_for_indexing.py

```diff
@@ -0,0 +1,42 @@
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("zerver", "0394_alter_realm_want_advertise_in_communities_directory"),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            sql=r"""
+            CREATE OR REPLACE FUNCTION extract_anchor_text(message_content text) RETURNS SETOF text IMMUTABLE LANGUAGE 'sql' AS $$
+                select anchor_text_match[1] as anchor_text from REGEXP_MATCHES(
+                    message_content,'<a [^>]+>([^<]+)<\/a>','g'
+                ) as anchor_text_match
+            $$ ;
+
+            CREATE OR REPLACE FUNCTION replace_url_delimiters(anchor_text text, replacement text) RETURNS text IMMUTABLE LANGUAGE 'sql' AS $$
+                select regexp_replace(anchor_text,'[:\/.?#@]', replacement, 'g');
+            $$ ;
+
+            CREATE OR REPLACE FUNCTION replace_anchor_text_url_delimiters(message_content text, replacement text) RETURNS text AS
+            $BODY$
+                DECLARE
+                    anchor_text text;
+                    result text;
+                BEGIN
+                    result := message_content;
+                    FOR anchor_text IN select extract_anchor_text(message_content)
+                    LOOP
+                        result := replace(result,  '>'|| anchor_text ||'<' , '>' || replace_url_delimiters(anchor_text, replacement) || '<');
+                    END LOOP;
+                RETURN result;
+                END;
+            $BODY$ LANGUAGE plpgsql;
+ 
+            CREATE OR REPLACE FUNCTION process_text_for_search(text_to_process text) RETURNS text IMMUTABLE LANGUAGE 'sql' AS $$
+                select replace_anchor_text_url_delimiters(text_to_process, ' ')
+            $$ ;
```

Comment:
I am working on related issue #22672 for PGroonga.

design question: why did you implement process_text_for_search() as a PostgreSQL function, instead of a Python function?  IMHO Python has more/better libraries for text parsing and is easier to test/debug ?

Reply:
Sorry for lack of formatting. Typing from mobile device
@asah because for running python functions we will need to use some extension and some zulip customers dont have superuser role to run `create extension` command needed for python procedural language in postgres. Let me dig some zulip chat thread where we discussed a bit about this later when i get back to my machine